tests/Controller/Api/External/ConfigControllerTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the mangel.io project.
+ *
+ * (c) Florian Moser <git@famoser.ch>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Controller\Api\External;
+
+use App\Service\Interfaces\PathServiceInterface;
+use App\Tests\Controller\Api\External\Base\ApiController;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class ConfigControllerTest extends ApiController
+{
+    /**
+     * @var string
+     */
+    private $configFilePath;
+
+    /**
+     * ConfigControllerTest constructor.
+     *
+     * @param string|null $name
+     * @param array $data
+     * @param string $dataName
+     */
+    public function __construct(?string $name = null, array $data = [], string $dataName = '')
+    {
+        parent::__construct($name, $data, $dataName);
+
+        self::bootKernel();
+        $pathService = self::$container->get(PathServiceInterface::class);
+
+        $this->configFilePath = $pathService->getTransientFolderRoot() . \DIRECTORY_SEPARATOR . 'domainOverrides.json';
+    }
+
+    /**
+     * tests the login functionality.
+     */
+    public function testDomainOverrides_configExists_returnsConfig()
+    {
+        $client = static::createClient();
+        $this->createConfigFile();
+
+        $response = $this->doConfigRequest($client);
+        $trialResponse = json_decode($response->getContent());
+        $this->assertNotNull($trialResponse->domainOverrides);
+
+        $domainOverrides = $trialResponse->domainOverrides;
+        $this->assertTrue(\is_array($domainOverrides));
+        $this->assertNotEmpty($domainOverrides);
+
+        $entry = $domainOverrides[0];
+        $this->assertNotNull($entry->userInputDomain);
+        $this->assertNotNull($entry->serverUrl);
+        $this->assertNotNull($entry->userLoginDomain);
+
+        $this->removeConfigFile();
+    }
+
+    /**
+     * tests the login functionality.
+     */
+    public function testDomainOverrides_configDoesNotExist_returnsNotFound()
+    {
+        $client = static::createClient();
+        $this->removeConfigFile();
+
+        $this->expectException(NotFoundHttpException::class);
+        $this->doConfigRequest($client);
+    }
+
+    private function createConfigFile()
+    {
+        file_put_contents($this->configFilePath, '{
+  "domainOverrides": [
```

Comment:
The only way this is ever accessed is when you have a specific `userInputDomain` and want to know the `serverURL` and `userLoginDomain`, so I'd implement this as a dictionary instead, which also more accurately reflects the design we originally agreed upon. See [my temporary local version](https://github.com/mangelio/iOS/blob/master/Issue%20Manager/Issue%20Manager/domains.public.json) for how I defined it. This naming is more accurate than in that version, so that's probably worth keeping.

Reply:
The dictionary is useful for the lookup; logn access an so on; but I'd rather have objects: I do not like to use dictionaries in JSON as you can not distinguish between an object with really weird property names and one that is actually supposed to be a dictionary.

Reply:
The distinction between object and dictionary is exactly what our API spec is for, no? Either way, the best point I see for this object array format is that the keys are explicitly labeled rather than just keying the objects. I'm honestly fine with either way, so although I'd personally prefer the dictionary format, I'll leave it up to you ¯\_(ツ)_/¯